concent_api/core/views.py

```diff
@@ -1241,6 +1209,712 @@ def store_message_and_message_status(
+                client_public_key   = subtask.provider.public_key_bytes,
+                queue               = PendingResponse.PendingResponseType.Receive,
+                subtask             = subtask,
+            )
+            store_pending_message(
+                response_type       = PendingResponse.ResponseType.ForceSubtaskResultsSettled,
+                client_public_key   = subtask.requestor.public_key_bytes,
+                queue               = PendingResponse.PendingResponseType.ReceiveOutOfBand,
+                subtask             = subtask,
+            )
+        # TODO when last use case will be implemented
+        # elif subtask.state == Subtask.SubtaskState.ADDITIONAL_VERIFICATION:
+
+
+def update_subtask_state(
+    subtask,
+    state,
+):
+    logging.log_change_subtask_state_name(
+        subtask.state,
+        state,
+    )
+    subtask.state = state
+    subtask.full_clean()
+    subtask.save()
+
+
+def store_pending_message(
+    response_type       = None,
+    client_public_key   = None,
+    queue               = None,
+    subtask             = None,
+    payment_message     = None,
+):
+    client = Client.objects.get_or_create_full_clean(client_public_key)
+    if payment_message is not None:
+        payment_committed_message = PaymentCommittedMessage(
+            payment_ts                  = datetime.datetime.fromtimestamp(payment_message.payment_ts, timezone.utc),
+            task_owner_key_bytes        = decode_key(payment_message.task_owner_key),
+            provider_eth_account_bytes  = payment_message.provider_eth_account,
+            amount_paid                 = payment_message.amount_paid,
+            recipient_type              = payment_message.recipient_type.name,  # pylint: disable=no-member
+            amount_pending              = payment_message.amount_pending
+        )
+        payment_committed_message.full_clean()
+        payment_committed_message.save()
+        receive_queue = PendingResponse(
+            response_type   = response_type.name,
+            client          = client,
+            queue           = queue.name,
+            subtask         = subtask,
+        )
+        receive_queue.payment_message = payment_committed_message
+    else:
+        receive_queue = PendingResponse(
+            response_type   = response_type.name,
+            client          = client,
+            queue           = queue.name,
+            subtask         = subtask,
+        )
+        logging.log_new_pending_response(
+            response_type.name,
+            queue.name,
+            subtask.subtask_id,
+            client.public_key,
+        )
+    receive_queue.full_clean()
+    receive_queue.save()
+
+
+def handle_messages_from_database(
+    client_public_key:  bytes   = None,
+    message_inbox_type: Enum    = None,
+):
+    assert client_public_key    not in ['', None]
+    assert message_inbox_type   in PendingResponse.PendingResponseType
```

Comment:
This could be a type annotation.

Reply:
As described in https://github.com/golemfactory/concent/pull/158, mypy in reporing an error if subclass of Enum is used.